app/Models/Extensions/PhotoCast.php

```diff
@@ -23,166 +58,75 @@ public function toReturnArray(): array
 	 *
-	 * @return array returns photo-attributes in a normalized structure
-	 */
-	public function prepareLocationData()
-	{
-		// Init
-		$photo = [];
-
-		// Set unchanged attributes
-		$photo['id'] = strval($this->id);
-		$photo['title'] = $this->title;
-		$photo['album'] = $this->album_id !== null ? strval($this->album_id) : null;
-		$photo['latitude'] = $this->latitude;
-		$photo['longitude'] = $this->longitude;
-
-		// if this is a video
-		if (strpos($this->type, 'video') === 0) {
-			$photoUrl = $this->thumbUrl;
-		} else {
-			$photoUrl = $this->url;
-		}
-
-		$photoUrl2x = '';
-		if ($photoUrl !== '') {
-			$photoUrl2x = Helpers::ex2x($photoUrl);
-		}
-
-		if ($this->small != '') {
-			$photo['small'] = Storage::url('small/' . $photoUrl);
-		} else {
-			$photo['small'] = '';
-		}
-
-		if ($this->small2x != '') {
-			$photo['small2x'] = Storage::url('small/' . $photoUrl2x);
-		} else {
-			$photo['small2x'] = '';
-		}
-
-		// Parse paths
-		$photo['thumbUrl'] = Storage::url('thumb/' . $this->thumbUrl);
-
-		if ($this->thumb2x == '1') {
-			$thumbUrl2x = Helpers::ex2x($this->thumbUrl);
-			$photo['thumb2x'] = Storage::url('thumb/' . $thumbUrl2x);
-		} else {
-			$photo['thumb2x'] = '';
-		}
-
-		$path_prefix = $this->type == 'raw' ? 'raw/' : 'big/';
-		$photo['url'] = Storage::url($path_prefix . $this->url);
-
-		if (isset($this->takestamp) && $this->takestamp != null) {
-			// Use takestamp
-			$photo['takedate'] = $this->takestamp->format('d F Y \a\t H:i');
-		} else {
-			$photo['takedate'] = '';
-		}
-
-		return $photo;
-	}
-
-	/**
-	 * Given a photo, return the proper URL.
+	 * @param string      $sizeVariant The name of the size variant which is being serialized; used to determine the correct path prefix
+	 * @param string|null $fileName    The filename
+	 * @param int|null    $width       The width of this variant
+	 * @param int|null    $height      The height of this variant
+	 *
+	 * @return array|null An associative array with the following attributes "url", "width" and "height" or null, if
+	 *                    any of the parameters is null
 	 */
-	public function urls(array &$return): void
+	protected function serializeSizeVariant(string $sizeVariant, ?string $fileName, ?int $width, ?int $height): ?array
 	{
-		// if this is a video
-		if (strpos($this->type, 'video') === 0) {
-			$photoUrl = $this->thumbUrl;
-
-			// We need to format the framerate (stored as focal) -> max 2 decimal digits
-			$return['focal'] = round($return['focal'], 2);
-		} elseif ($this->type == 'raw') {
-			// It's a raw file -> we also use jpeg as extension
-			$photoUrl = $this->thumbUrl;
-		} else {
-			$photoUrl = $this->url;
-		}
-		$photoUrl2x = ($photoUrl !== '') ? Helpers::ex2x($photoUrl) : '';
-
-		$sizes = [
-			'medium' => $photoUrl,
-			'medium2x' => $photoUrl2x,
-			'small' => $photoUrl,
-			'small2x' => $photoUrl2x,
+		return ($width === null || $height === null || $fileName === null || $fileName === '') ? null : [
```

Comment:
I would prefer the use of a `if` instead of a ternary here as this improve the readability (Nitpicking).

I have nothing against ternary in general. :)

Reply:
Fixed in commit c513d92d.